doc/src/modules/physics/mechanics/examples/four_bar_linkage_example.rst

```diff
@@ -80,3 +81,67 @@ using :class:`~.KanesMethod` as the backend. ::
 
+Revealing noncontributing forces
+--------------------------------
+
+To reveal the noncontributing forces at the closing joint, we require to
+introduce auxiliary speeds in the x and y-direction at the endpoint.
+
+   >>> uaux1, uaux2 = dynamicsymbols('uaux1:3')
+   >>> end_point_aux = end_point.locatenew('end_point_aux', 0)
+   >>> end_point_aux.set_vel(N, end_point.vel(N) + uaux1*N.x + uaux2*N.y)
+
+To ensure that speeds are included in the velocity constraints, we must manually
+overwrite the velocity constraints because those are by default specified as the
+time derivatives of the holonomic constraints.
+
+   >>> system.velocity_constraints = [
+   ...    end_point_aux.vel(N).dot(N.x), end_point_aux.vel(N).dot(N.y)]
+
+When adding the noncontributing forces we need them to depend only on the
+auxiliary velocity and not the velocity that is eliminated by the constraints.
+This can be achieved by applying an equal and opposite force to the
+non-auxiliary endpoint.
+
+   >>> faux1, faux2 = dynamicsymbols('faux1:3')
+   >>> noncontributing_forces = [
+   ...   Force(end_point_aux, faux1*N.x + faux2*N.y),
+   ...   Force(end_point, -(faux1*N.x + faux2*N.y)),
+   ... ]
+
+Alternatively, we can specify a new point that already subtracts the velocity
+eliminated by the constraints.
+
+   >>> end_point_forces = end_point.locatenew('end_point_forces', 0)
+   >>> end_point_forces.set_vel(N, uaux1*N.x + uaux2*N.y)
+   >>> noncontributing_forces = [Force(end_point_forces, faux1*N.x + faux2*N.y)]
+
+Next, we can add the auxiliary speeds and noncontributing forces to the system.
+
+   >>> system.add_loads(*noncontributing_forces)
+   >>> system.u_aux = [uaux1, uaux2]
+
+To include the gravity we can use :meth:`~.System.apply_uniform_gravity` before
+validating the system and forming the equations of motion.
+
+   >>> g = symbols('g')
+   >>> system.apply_uniform_gravity(-g*N.y)
+   >>> system.validate_system()
+   >>> eoms = system.form_eoms()
+
+With the equations of motion formed we can solve the auxiliary equations for the
+noncontributing forces and compute their values for a simple configuration.
+
+   >>> auxiliary_eqs = system.eom_method.auxiliary_eqs
+   >>> forces_eqs = Matrix.LUsolve(
+   ...   *linear_eq_to_matrix(auxiliary_eqs, [faux1, faux2]))
+   >>> subs = {
+   ...   l1: 2, l2: 1, l3: 2, l4: 1,
+   ...   rho: 5, g: 9.81,
+   ...   q1: pi/2, q2: pi/2, q3: pi/2,
+   ...   u1: 0, u2: 0, u3: 0, u1.diff(): 0, u2.diff(): 0, u3.diff(): 0,
+   ... }
+   >>> forces_eqs.xreplace(subs)
+   Matrix([
+   [    0],
+   [-98.1]])
```

Comment:
Does that value look right? Magnitude an direction? It should be the proportion of weight on one end of how ever it is configured?

Reply:
The four-bar linkage is oriented as a perfect rectangle with link2 and link4 oriented upwards. Therefore, half of the mass of link3, `0.5 * 2 * 5 = 5`, and the full mass of link4, `1 * 5 = 5` rests on this joint. So a force of `9.81 * (5 + 5) = 98.1` acts downwards on the closing joint.